src/include/migraphx/op/pointwise.hpp

```diff
@@ -45,23 +45,18 @@ struct pointwise
         {
             MIGRAPHX_THROW("should have one submodule.");
         }
-        auto* pm = mods.front();
-        if(pm->get_output_shapes().size() != 1)
-            MIGRAPHX_THROW("pointwise should have only one output.");
         if(inputs.empty())
             MIGRAPHX_THROW("pointwise should have at least one input");
+        auto* pm    = mods.front();
         auto pnames = pm->get_parameter_names();
-        std::sort(pnames.begin(), pnames.end());
         check_shapes{inputs, *this}.has(pnames.size()).same_dims();
 
-        auto type = pm->get_output_shapes().front().type();
-
-        // Scalar output if all inputs are scalar
-        if(inputs.front().elements() == 1 and
-           all_of(inputs, [](const auto& s) { return s.scalar(); }))
-            return shape{type};
-
-        return shape::from_permutation(type, inputs.front().lens(), find_permutation(inputs));
+        auto result = pm->compute_shapes(
+            inputs,
+            {.name = name(), .strict_type = true, .scalar_const_out_lens = inputs.front().lens()});
```

Comment:
Can you add some tests inside `op_shape_test.cpp` for different cases for the pointwise op ? 

e.g. 
with mutiple outputs, 
with one of the outputs being scalar, 
with strict_type being true and throwing

Reply:
All the inputs to pointwise should have the dimensions. We need to create a dimension for the scalar literals in the pointwise module(which only have a single element) so we can compute the shape correctly.

Reply:
> Can you add some tests inside op_shape_test.cpp for different cases for the pointwise op ?

You can't test operators like pointwise op in op_shape_test.cpp.

>  with mutiple outputs,

There is already a ref test for this.

>  with one of the outputs being scalar,

I am not sure what you are intending for such a test. This wont get coverage for `scalar_const_out_lens` flag. Either way both test cases(ie scalar output and `scalar_const_out_lens` flag) are already in test/fuse_pointwise.cpp.

Reply:
> You can, we already have some.

Ah I missed that. I will add a test for the `.strict_type` flag. 

> I was looking for test like where it would use .scalar_const_out_lens flag.

Yea we already have that test in test/fuse_pointwise.cpp:

https://github.com/ROCm/AMDMIGraphX/blob/bd52f7172abb76722283fcb9660990cf94db7190/test/fuse_pointwise.cpp#L221

If we dont use the `.scalar_const_out_lens` flag then this will throw an error because the inputs to the `add` operator will be `{2, 3}` and a scalar. 

> > with one of the outputs being scalar,

We already have a test for this as well in test/fuse_pointwise.cpp:

https://github.com/ROCm/AMDMIGraphX/blob/bd52f7172abb76722283fcb9660990cf94db7190/test/fuse_pointwise.cpp#L305

It takes all scalar inputs and checks that we have a scalar output.

Reply:
Test added.